compiler/rustc_codegen_ssa/src/mir/block.rs

```diff
@@ -773,7 +773,11 @@ impl<'a, 'tcx, Bx: BuilderMethods<'a, 'tcx>> FunctionCx<'a, 'tcx, Bx> {
         };
         let def = instance.map(|i| i.def);
 
-        if let Some(ty::InstanceDef::DropGlue(_, None)) = def {
+        if let Some(
+            ty::InstanceDef::DropGlue(_, None)
+            | ty::InstanceDef::AsyncDropGlueCtorShim(_, None)
+            | ty::InstanceDef::AsyncDropGlue(_, None),
+        ) = def {
```

Comment:
Can you make this change to rustc_codegen_cranelift too at https://github.com/rust-lang/rust/blob/f092f73c112914752d3290fbbbfa2271c78f1088/compiler/rustc_codegen_cranelift/src/abi/mod.rs#L600 and https://github.com/rust-lang/rust/blob/f092f73c112914752d3290fbbbfa2271c78f1088/compiler/rustc_codegen_cranelift/src/abi/mod.rs#L415?